test/testenv.py

```diff
@@ -61,6 +61,7 @@ def get_clang_args(self):
 
 class Testcase:
     _options_schema = strictyaml.Map({
+        strictyaml.Optional('test'): strictyaml.Seq(strictyaml.Str()),
```

Comment:
I think it should be possible to make this a Seq of Enum, so that only valid strings are accepted. Otherwise you could add "foo" and it would pass without matching any of the tests. (Can be fixed later too.)

Hate to nitpick, but "test" is truly too overloaded here. (But again, trivial to fix afterwards if we come up with a better name.)

Reply:
On the key name... I get your point. Would you feel better about `test-class` or something along those lines?